decam_process.py

```diff
@@ -15,324 +15,378 @@
+        # feed everything to the argument parser
+        parsedCmd = argumentParser.parse_args(config=config, args=args)
+        # finally, run the ingestTask
+        ingestTask.run(parsedCmd)
+        # Copy ref_cats files to repo (needed for doProcessCcd)
+        gaiatarball = os.path.join(ref_cats, 'gaia_HiTS_2015.tar.gz')
+        panstarrstarball = os.path.join(ref_cats, 'ps1_HiTS_2015.tar.gz')
+        tarfile.open(gaiatarball, 'r').extractall(os.path.join(repo, 'ref_cats', 'gaia'))
+        tarfile.open(panstarrstarball, 'r').extractall(os.path.join(repo, 'ref_cats', 'pan-starrs'))
+        print('Images are now ingested in {0}'.format(repo))
     return
 
 
-def doIngestCalibs(repo, calibrepo, datafiles):
+def doIngestCalibs(repo, calibrepo, calibdatafiles, defectfiles):
     '''
-    Ingest DECam MasterCal biases and flats into a calibration repository with a corresponding registry
-
-    Darks (and fringes, if applicable) must be ingested manually! e.g.,
-    $ cd calibrepo
-    $ ingestCalibs.py ../repo --calib . --calibType defect --validity 999
-          ../HiTS/MasterCals/defects/2014-12-05/*fits
-
-    USAGE:
-    $ python hits_ingest.py ingestCalibs -f path/to/calibrations/somefiles*.fits.fz
+    Ingest DECam MasterCal biases and flats into a calibration repository with a corresponding registry.
+    Also ingest DECam defects into the calib registry.
+
+    Parameters
+    ----------
+    repo: `str`
+        The output repository location on disk where ingested raw images live.
+    calibrepo: `str`
+        The output repository location on disk where ingested calibration images live.
+    calibdatafiles: `list`
+        A list of the filenames of each flat and bias image file.
+    defectfiles: `list`
+            A list of the filenames of each defect image file.
+            The first element in this list must be the name of a .tar.gz file
+            which contains all the compressed defect images.
 
     BASH EQUIVALENT:
-    $ ingestCalibs.py repo --calib calibrepo --mode=link --validity 999 datafiles
+    $ ingestCalibs.py repo --calib calibrepo --mode=link --validity 999 calibdatafiles
+    $ cd calibrepo
+    $ ingestCalibs.py ../../repo --calib . --mode=skip --calibType defect --validity 999 defectfiles
+    $ cd ..
 
     RESULT:
-    calibrepo populated with *links* to files in datadir,
-    organized by date (bias/zero and flat images only)
-    sqlite3 database registry of ingested calibration products
-    created in calibrepo
+    calibrepo populated with *links* to calibdatafiles,
+    organized by date (bias and flat images only)
+    sqlite3 database registry of ingested calibration products (bias, flat,
+    and defect images) created in calibrepo
+
+    NOTE:
+    calib ingestion ingests *all* the calibs, not just the ones needed
+    for the specified visits. We may want to revisit this in the future.
     '''
+
+    def flatBiasIngest(repo, calibrepo, calibdatafiles):
```

Comment:
This function would be nice to have at the module level instead of as an internal function. Same for `defectIngest`.